edb/server/compiler/errormech.py

```diff
@@ -252,3 +270,72 @@ def interpret_backend_error(schema, fields):
         return errors.TransactionDeadlockError(message)
 
     return errors.InternalServerError(message)
+
+
+def interpret_backend_error(schema, fields):
+    err_details = get_error_details(fields)
+    if isinstance(err_details, errors.EdgeDBError):
+        return err_details
+
+    message = err_details.message
+    code = err_details.code
+    schema_name = err_details.schema_name
+    table_name = err_details.table_name
+    column_name = err_details.column_name
+    constraint_name = err_details.constraint_name
+
+    if code == PGError.NotNullViolationError:
+        source_name = pointer_name = None
+
+        if schema_name and table_name:
+            tabname = (schema_name, table_name)
+
+            source = common.get_object_from_backend_name(
+                schema, s_objtypes.ObjectType, tabname)
+            source_name = source.get_displayname(schema)
+
+            if column_name:
+                pointer_name = column_name
+
+        if pointer_name is not None:
+            pname = f'{source_name}.{pointer_name}'
+
+            return errors.MissingRequiredError(
+                f'missing value for required property {pname}')
+
+        else:
+            return errors.InternalServerError(message)
+
+    elif code in constraint_errors:
+        source = None
+
+        for errtype, ere in constraint_res.items():
+            m = ere.match(message)
+            if m:
+                error_type = errtype
+                break
+        # no need for else clause since it would have been handled by
```

Comment:
You have a potential `NameError` for `error_type` here.

Reply:
OK, I've added an initialization for `error_type` just in case. Incidentally I removed source from this code block since it's not used (but oddly wasn't flagged as unused by flake).